src/parser/chunkparser.hpp

```diff
@@ -0,0 +1,61 @@
+#pragma once
+#include <array>
+#include <nlohmann/json.hpp>
+
+namespace docgen {
+
+//////////////////////////////////////////////////////
+// ChunkParser Declaration
+//////////////////////////////////////////////////////
+
+// ChunkParser is a class that will parse a chunk of text into JSON format.
+class ChunkParser
+{
+public:
+
+    // Main function that user will call to parse a chunk of text.
+    // This function will update the json object with information extracted from chunk.
+    void parse(const char *begin, 
+               const char *end);
+
+private:
+
+    // stores parsed information
+    nlohmann::json json;
```

Comment:
Call something more descriptive maybe? `parsed_` is an idea, but anything like that.